### PR TITLE
Adding unpartitioned size attribute

### LINKED ISSUE #709: Allow unpartitioned free space for disk images (oem, vmx)
## Description

Currently it is possible to set the size of the disk in the description file using the `<size>` tag and Kiwi partitions all disk space available. In case a bigger disk was needed with some unpartitioned free space the only way to achieve it was using the `kiwi image resize` to expand the built disk, thus this was not part of the build process neither part of the image description.

It is suggested to extend KIWI to allow the configuration of unpartitioned space in the disk configuration within the image description and also as part of the build process. Could be described as:

```xml
<size unit="G" unpartitioned="4G">10</size>
```
leading to a disk of 10G with 4G unpartitioned at the end of the disk. It can be implemented by reusing resize code once the image is already created.

This extension is related to https://github.com/suse/kiwi-descriptions/pull/21 and the fate [323874](https://fate.suse.com/323874)

### PR DESCRIPTION
**Pull request by davidcassany**
This commit adds the possibility of setting some unpartitioned area
after the systemdisl partition into the image.

Fixes #709

### CODE REVIEW

File: kiwi/xml_state.py [hunk 1]
```diff
@@ -728,6 +734,25 @@ def get_build_type_size(self):
...
+        if size_section:
+            unit = size_section[0].get_unit()
+            unpartitioned = size_section[0].get_unpartitioned()
+            if unpartitioned is None:
+                unpartitioned = 0
+            if unit == 'G':
+                unpartitioned *= 1024
+            return unpartitioned
+        return 0
+
```
[the remark this answers is not in the document]
Comment (davidcassany) [reply kiwi/xml_state.py]: Right agree we can return bytes here and use the StringToSize method.

File: kiwi/schema/kiwi.rnc [hunk 2]
```diff
@@ -1066,12 +1066,27 @@ div {
...
+        ## Specifies the image empty space that will not be partitioned
+        attribute unpartitioned { xsd:nonNegativeInteger }
+    sch:pattern [
+        id = "unpartitioned_validation"
+        sch:rule [
+            context = "type/size[@unpartitioned]"
+            sch:assert [
+                test = "@unpartitioned <= text()"
+                "unpartitioned size must be smaller or equal to the defined "~
+                "size"
```
[the remark this answers is not in the document]
Comment (davidcassany) [reply kiwi/schema/kiwi.rnc]: I included it here thinking it was needed for the line break, as we have it in the other `<type>` patterns. Just did a quick test and I can't see any difference, either in this one or in the ones we had. Probably some tool changed in between, I'll do some little more testing and remove them all if I can't see any difference.

[the remark this answers is not in the document]
Comment (schaefi) [reply kiwi/schema/kiwi.rnc]: yes this was relevant for docbook and from the times when Thomas tried to add doc chapters for docbook rendered documentation. For our use case no matter if oxygen or our own doc tool reads them those line breaks are of no interest

**Comment by davidcassany** [comment]
@schaefi, now it is reusing StringToSize and got rid of the "~" in schematron patterns.

**Comment by schaefi** [comment]
Looks great to me :+1:

**Pull request merged**